File: problems/0007_reverse-integer/solution.js
/**
 * @param {number} x
 * @return {number}
 */
var reverse = function (x) {
    'use strict'
    const sign = Math.sign(x)
    x *= sign
    let result = 0
    while (x != 0) {
        const r = x % 10
        result = result * 10 + r
        x = (x - r) / 10
    }
    result *= sign
    if (result < -2147483648 || 2147483647 < result)
        return 0
    else
        return result
};

File: problems/0007_reverse-integer/solution.c
int reverse(int x){
    const int sign = (x < 0 ? -1 : 1);
    long n = ((long) x) * sign,
         result = 0;
    for (; n != 0; n /= 10)
        result = result * 10 + (n % 10);
    result *= sign;
    return (result < INT_MIN || INT_MAX < result ?
           0 : (int) result);
}
